src/com/elixr/training/Main.java

```diff
@@ -0,0 +1,65 @@
+package com.elixr.training;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.StringTokenizer;
+import java.lang.String;
+
+public class Main {
+
+    public static void main(String[] args) {
+        if (args.length != 2) {
+            System.out.println("Argument missing");
+            return;
+        }
+        String inputfilepath = args[0];
+        String searchword = args[1];
+        File file = new File(inputfilepath);
+        System.out.println("Processing................");
+        String data = readFile(inputfilepath);
+        isSupportedFile(file);
+        searchword(data, searchword);
+
+    }
+
+    public static String readFile(String fileName) {
+        String data = null;
+        try {
+            data = new String(Files.readAllBytes(Paths.get(fileName)));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return data;
+    }
+
+    public static void isSupportedFile(File file) {
+        if (file.getName().endsWith(".txt") && (file.length() != 0)) {
+            System.out.println("File format supported");
+        } else {
+            System.out.println("File format not supported");
+            System.exit(0);
```

Comment:
Do not use System.exit(), just a return a boolean to indicate whether the file is valid or not, and in the main method check for the return value and proceed if needed

Reply:
changed it to boolean and removed .exit method